<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="L Collado-Torres" />


<title>Example report using bumphunter results</title>


<style type="text/css">code{white-space: pre;}</style>
<link href="data:text/css,pre%20%2Eoperator%2C%0Apre%20%2Eparen%20%7B%0A%20color%3A%20rgb%28104%2C%20118%2C%20135%29%0A%7D%0A%0Apre%20%2Eliteral%20%7B%0A%20color%3A%20%23990073%0A%7D%0A%0Apre%20%2Enumber%20%7B%0A%20color%3A%20%23099%3B%0A%7D%0A%0Apre%20%2Ecomment%20%7B%0A%20color%3A%20%23998%3B%0A%20font%2Dstyle%3A%20italic%0A%7D%0A%0Apre%20%2Ekeyword%20%7B%0A%20color%3A%20%23900%3B%0A%20font%2Dweight%3A%20bold%0A%7D%0A%0Apre%20%2Eidentifier%20%7B%0A%20color%3A%20rgb%280%2C%200%2C%200%29%3B%0A%7D%0A%0Apre%20%2Estring%20%7B%0A%20color%3A%20%23d14%3B%0A%7D%0A" rel="stylesheet" type="text/css" />
<script src="data:application/x-javascript,%0Avar%20hljs%3Dnew%20function%28%29%7Bfunction%20m%28p%29%7Breturn%20p%2Ereplace%28%2F%26%2Fgm%2C%22%26amp%3B%22%29%2Ereplace%28%2F%3C%2Fgm%2C%22%26lt%3B%22%29%7Dfunction%20f%28r%2Cq%2Cp%29%7Breturn%20RegExp%28q%2C%22m%22%2B%28r%2EcI%3F%22i%22%3A%22%22%29%2B%28p%3F%22g%22%3A%22%22%29%29%7Dfunction%20b%28r%29%7Bfor%28var%20p%3D0%3Bp%3Cr%2EchildNodes%2Elength%3Bp%2B%2B%29%7Bvar%20q%3Dr%2EchildNodes%5Bp%5D%3Bif%28q%2EnodeName%3D%3D%22CODE%22%29%7Breturn%20q%7Dif%28%21%28q%2EnodeType%3D%3D3%26%26q%2EnodeValue%2Ematch%28%2F%5Cs%2B%2F%29%29%29%7Bbreak%7D%7D%7Dfunction%20h%28t%2Cs%29%7Bvar%20p%3D%22%22%3Bfor%28var%20r%3D0%3Br%3Ct%2EchildNodes%2Elength%3Br%2B%2B%29%7Bif%28t%2EchildNodes%5Br%5D%2EnodeType%3D%3D3%29%7Bvar%20q%3Dt%2EchildNodes%5Br%5D%2EnodeValue%3Bif%28s%29%7Bq%3Dq%2Ereplace%28%2F%5Cn%2Fg%2C%22%22%29%7Dp%2B%3Dq%7Delse%7Bif%28t%2EchildNodes%5Br%5D%2EnodeName%3D%3D%22BR%22%29%7Bp%2B%3D%22%5Cn%22%7Delse%7Bp%2B%3Dh%28t%2EchildNodes%5Br%5D%29%7D%7D%7Dif%28%2FMSIE%20%5B678%5D%2F%2Etest%28navigator%2EuserAgent%29%29%7Bp%3Dp%2Ereplace%28%2F%5Cr%2Fg%2C%22%5Cn%22%29%7Dreturn%20p%7Dfunction%20a%28s%29%7Bvar%20r%3Ds%2EclassName%2Esplit%28%2F%5Cs%2B%2F%29%3Br%3Dr%2Econcat%28s%2EparentNode%2EclassName%2Esplit%28%2F%5Cs%2B%2F%29%29%3Bfor%28var%20q%3D0%3Bq%3Cr%2Elength%3Bq%2B%2B%29%7Bvar%20p%3Dr%5Bq%5D%2Ereplace%28%2F%5Elanguage%2D%2F%2C%22%22%29%3Bif%28e%5Bp%5D%29%7Breturn%20p%7D%7D%7Dfunction%20c%28q%29%7Bvar%20p%3D%5B%5D%3B%28function%28s%2Ct%29%7Bfor%28var%20r%3D0%3Br%3Cs%2EchildNodes%2Elength%3Br%2B%2B%29%7Bif%28s%2EchildNodes%5Br%5D%2EnodeType%3D%3D3%29%7Bt%2B%3Ds%2EchildNodes%5Br%5D%2EnodeValue%2Elength%7Delse%7Bif%28s%2EchildNodes%5Br%5D%2EnodeName%3D%3D%22BR%22%29%7Bt%2B%3D1%7Delse%7Bif%28s%2EchildNodes%5Br%5D%2EnodeType%3D%3D1%29%7Bp%2Epush%28%7Bevent%3A%22start%22%2Coffset%3At%2Cnode%3As%2EchildNodes%5Br%5D%7D%29%3Bt%3Darguments%2Ecallee%28s%2EchildNodes%5Br%5D%2Ct%29%3Bp%2Epush%28%7Bevent%3A%22stop%22%2Coffset%3At%2Cnode%3As%2EchildNodes%5Br%5D%7D%29%7D%7D%7D%7Dreturn%20t%7D%29%28q%2C0%29%3Breturn%20p%7Dfunction%20k%28y%2Cw%2Cx%29%7Bvar%20q%3D0%3Bvar%20z%3D%22%22%3Bvar%20s%3D%5B%5D%3Bfunction%20u%28%29%7Bif%28y%2Elength%26%26w%2Elength%29%7Bif%28y%5B0%5D%2Eoffset%21%3Dw%5B0%5D%2Eoffset%29%7Breturn%28y%5B0%5D%2Eoffset%3Cw%5B0%5D%2Eoffset%29%3Fy%3Aw%7Delse%7Breturn%20w%5B0%5D%2Eevent%3D%3D%22start%22%3Fy%3Aw%7D%7Delse%7Breturn%20y%2Elength%3Fy%3Aw%7D%7Dfunction%20t%28D%29%7Bvar%20A%3D%22%3C%22%2BD%2EnodeName%2EtoLowerCase%28%29%3Bfor%28var%20B%3D0%3BB%3CD%2Eattributes%2Elength%3BB%2B%2B%29%7Bvar%20C%3DD%2Eattributes%5BB%5D%3BA%2B%3D%22%20%22%2BC%2EnodeName%2EtoLowerCase%28%29%3Bif%28C%2Evalue%21%3D%3Dundefined%26%26C%2Evalue%21%3D%3Dfalse%26%26C%2Evalue%21%3D%3Dnull%29%7BA%2B%3D%27%3D%22%27%2Bm%28C%2Evalue%29%2B%27%22%27%7D%7Dreturn%20A%2B%22%3E%22%7Dwhile%28y%2Elength%7C%7Cw%2Elength%29%7Bvar%20v%3Du%28%29%2Esplice%280%2C1%29%5B0%5D%3Bz%2B%3Dm%28x%2Esubstr%28q%2Cv%2Eoffset%2Dq%29%29%3Bq%3Dv%2Eoffset%3Bif%28v%2Eevent%3D%3D%22start%22%29%7Bz%2B%3Dt%28v%2Enode%29%3Bs%2Epush%28v%2Enode%29%7Delse%7Bif%28v%2Eevent%3D%3D%22stop%22%29%7Bvar%20p%2Cr%3Ds%2Elength%3Bdo%7Br%2D%2D%3Bp%3Ds%5Br%5D%3Bz%2B%3D%28%22%3C%2F%22%2Bp%2EnodeName%2EtoLowerCase%28%29%2B%22%3E%22%29%7Dwhile%28p%21%3Dv%2Enode%29%3Bs%2Esplice%28r%2C1%29%3Bwhile%28r%3Cs%2Elength%29%7Bz%2B%3Dt%28s%5Br%5D%29%3Br%2B%2B%7D%7D%7D%7Dreturn%20z%2Bm%28x%2Esubstr%28q%29%29%7Dfunction%20j%28%29%7Bfunction%20q%28x%2Cy%2Cv%29%7Bif%28x%2Ecompiled%29%7Breturn%7Dvar%20u%3Bvar%20s%3D%5B%5D%3Bif%28x%2Ek%29%7Bx%2ElR%3Df%28y%2Cx%2El%7C%7Chljs%2EIR%2Ctrue%29%3Bfor%28var%20w%20in%20x%2Ek%29%7Bif%28%21x%2Ek%2EhasOwnProperty%28w%29%29%7Bcontinue%7Dif%28x%2Ek%5Bw%5D%20instanceof%20Object%29%7Bu%3Dx%2Ek%5Bw%5D%7Delse%7Bu%3Dx%2Ek%3Bw%3D%22keyword%22%7Dfor%28var%20r%20in%20u%29%7Bif%28%21u%2EhasOwnProperty%28r%29%29%7Bcontinue%7Dx%2Ek%5Br%5D%3D%5Bw%2Cu%5Br%5D%5D%3Bs%2Epush%28r%29%7D%7D%7Dif%28%21v%29%7Bif%28x%2EbWK%29%7Bx%2Eb%3D%22%5C%5Cb%28%22%2Bs%2Ejoin%28%22%7C%22%29%2B%22%29%5C%5Cs%22%7Dx%2EbR%3Df%28y%2Cx%2Eb%3Fx%2Eb%3A%22%5C%5CB%7C%5C%5Cb%22%29%3Bif%28%21x%2Ee%26%26%21x%2EeW%29%7Bx%2Ee%3D%22%5C%5CB%7C%5C%5Cb%22%7Dif%28x%2Ee%29%7Bx%2EeR%3Df%28y%2Cx%2Ee%29%7D%7Dif%28x%2Ei%29%7Bx%2EiR%3Df%28y%2Cx%2Ei%29%7Dif%28x%2Er%3D%3D%3Dundefined%29%7Bx%2Er%3D1%7Dif%28%21x%2Ec%29%7Bx%2Ec%3D%5B%5D%7Dx%2Ecompiled%3Dtrue%3Bfor%28var%20t%3D0%3Bt%3Cx%2Ec%2Elength%3Bt%2B%2B%29%7Bif%28x%2Ec%5Bt%5D%3D%3D%22self%22%29%7Bx%2Ec%5Bt%5D%3Dx%7Dq%28x%2Ec%5Bt%5D%2Cy%2Cfalse%29%7Dif%28x%2Estarts%29%7Bq%28x%2Estarts%2Cy%2Cfalse%29%7D%7Dfor%28var%20p%20in%20e%29%7Bif%28%21e%2EhasOwnProperty%28p%29%29%7Bcontinue%7Dq%28e%5Bp%5D%2EdM%2Ce%5Bp%5D%2Ctrue%29%7D%7Dfunction%20d%28B%2CC%29%7Bif%28%21j%2Ecalled%29%7Bj%28%29%3Bj%2Ecalled%3Dtrue%7Dfunction%20q%28r%2CM%29%7Bfor%28var%20L%3D0%3BL%3CM%2Ec%2Elength%3BL%2B%2B%29%7Bif%28%28M%2Ec%5BL%5D%2EbR%2Eexec%28r%29%7C%7C%5Bnull%5D%29%5B0%5D%3D%3Dr%29%7Breturn%20M%2Ec%5BL%5D%7D%7D%7Dfunction%20v%28L%2Cr%29%7Bif%28D%5BL%5D%2Ee%26%26D%5BL%5D%2EeR%2Etest%28r%29%29%7Breturn%201%7Dif%28D%5BL%5D%2EeW%29%7Bvar%20M%3Dv%28L%2D1%2Cr%29%3Breturn%20M%3FM%2B1%3A0%7Dreturn%200%7Dfunction%20w%28r%2CL%29%7Breturn%20L%2Ei%26%26L%2EiR%2Etest%28r%29%7Dfunction%20K%28N%2CO%29%7Bvar%20M%3D%5B%5D%3Bfor%28var%20L%3D0%3BL%3CN%2Ec%2Elength%3BL%2B%2B%29%7BM%2Epush%28N%2Ec%5BL%5D%2Eb%29%7Dvar%20r%3DD%2Elength%2D1%3Bdo%7Bif%28D%5Br%5D%2Ee%29%7BM%2Epush%28D%5Br%5D%2Ee%29%7Dr%2D%2D%7Dwhile%28D%5Br%2B1%5D%2EeW%29%3Bif%28N%2Ei%29%7BM%2Epush%28N%2Ei%29%7Dreturn%20f%28O%2CM%2Ejoin%28%22%7C%22%29%2Ctrue%29%7Dfunction%20p%28M%2CL%29%7Bvar%20N%3DD%5BD%2Elength%2D1%5D%3Bif%28%21N%2Et%29%7BN%2Et%3DK%28N%2CE%29%7DN%2Et%2ElastIndex%3DL%3Bvar%20r%3DN%2Et%2Eexec%28M%29%3Breturn%20r%3F%5BM%2Esubstr%28L%2Cr%2Eindex%2DL%29%2Cr%5B0%5D%2Cfalse%5D%3A%5BM%2Esubstr%28L%29%2C%22%22%2Ctrue%5D%7Dfunction%20z%28N%2Cr%29%7Bvar%20L%3DE%2EcI%3Fr%5B0%5D%2EtoLowerCase%28%29%3Ar%5B0%5D%3Bvar%20M%3DN%2Ek%5BL%5D%3Bif%28M%26%26M%20instanceof%20Array%29%7Breturn%20M%7Dreturn%20false%7Dfunction%20F%28L%2CP%29%7BL%3Dm%28L%29%3Bif%28%21P%2Ek%29%7Breturn%20L%7Dvar%20r%3D%22%22%3Bvar%20O%3D0%3BP%2ElR%2ElastIndex%3D0%3Bvar%20M%3DP%2ElR%2Eexec%28L%29%3Bwhile%28M%29%7Br%2B%3DL%2Esubstr%28O%2CM%2Eindex%2DO%29%3Bvar%20N%3Dz%28P%2CM%29%3Bif%28N%29%7Bx%2B%3DN%5B1%5D%3Br%2B%3D%27%3Cspan%20class%3D%22%27%2BN%5B0%5D%2B%27%22%3E%27%2BM%5B0%5D%2B%22%3C%2Fspan%3E%22%7Delse%7Br%2B%3DM%5B0%5D%7DO%3DP%2ElR%2ElastIndex%3BM%3DP%2ElR%2Eexec%28L%29%7Dreturn%20r%2BL%2Esubstr%28O%2CL%2Elength%2DO%29%7Dfunction%20J%28L%2CM%29%7Bif%28M%2EsL%26%26e%5BM%2EsL%5D%29%7Bvar%20r%3Dd%28M%2EsL%2CL%29%3Bx%2B%3Dr%2Ekeyword%5Fcount%3Breturn%20r%2Evalue%7Delse%7Breturn%20F%28L%2CM%29%7D%7Dfunction%20I%28M%2Cr%29%7Bvar%20L%3DM%2EcN%3F%27%3Cspan%20class%3D%22%27%2BM%2EcN%2B%27%22%3E%27%3A%22%22%3Bif%28M%2ErB%29%7By%2B%3DL%3BM%2Ebuffer%3D%22%22%7Delse%7Bif%28M%2EeB%29%7By%2B%3Dm%28r%29%2BL%3BM%2Ebuffer%3D%22%22%7Delse%7By%2B%3DL%3BM%2Ebuffer%3Dr%7D%7DD%2Epush%28M%29%3BA%2B%3DM%2Er%7Dfunction%20G%28N%2CM%2CQ%29%7Bvar%20R%3DD%5BD%2Elength%2D1%5D%3Bif%28Q%29%7By%2B%3DJ%28R%2Ebuffer%2BN%2CR%29%3Breturn%20false%7Dvar%20P%3Dq%28M%2CR%29%3Bif%28P%29%7By%2B%3DJ%28R%2Ebuffer%2BN%2CR%29%3BI%28P%2CM%29%3Breturn%20P%2ErB%7Dvar%20L%3Dv%28D%2Elength%2D1%2CM%29%3Bif%28L%29%7Bvar%20O%3DR%2EcN%3F%22%3C%2Fspan%3E%22%3A%22%22%3Bif%28R%2ErE%29%7By%2B%3DJ%28R%2Ebuffer%2BN%2CR%29%2BO%7Delse%7Bif%28R%2EeE%29%7By%2B%3DJ%28R%2Ebuffer%2BN%2CR%29%2BO%2Bm%28M%29%7Delse%7By%2B%3DJ%28R%2Ebuffer%2BN%2BM%2CR%29%2BO%7D%7Dwhile%28L%3E1%29%7BO%3DD%5BD%2Elength%2D2%5D%2EcN%3F%22%3C%2Fspan%3E%22%3A%22%22%3By%2B%3DO%3BL%2D%2D%3BD%2Elength%2D%2D%7Dvar%20r%3DD%5BD%2Elength%2D1%5D%3BD%2Elength%2D%2D%3BD%5BD%2Elength%2D1%5D%2Ebuffer%3D%22%22%3Bif%28r%2Estarts%29%7BI%28r%2Estarts%2C%22%22%29%7Dreturn%20R%2ErE%7Dif%28w%28M%2CR%29%29%7Bthrow%22Illegal%22%7D%7Dvar%20E%3De%5BB%5D%3Bvar%20D%3D%5BE%2EdM%5D%3Bvar%20A%3D0%3Bvar%20x%3D0%3Bvar%20y%3D%22%22%3Btry%7Bvar%20s%2Cu%3D0%3BE%2EdM%2Ebuffer%3D%22%22%3Bdo%7Bs%3Dp%28C%2Cu%29%3Bvar%20t%3DG%28s%5B0%5D%2Cs%5B1%5D%2Cs%5B2%5D%29%3Bu%2B%3Ds%5B0%5D%2Elength%3Bif%28%21t%29%7Bu%2B%3Ds%5B1%5D%2Elength%7D%7Dwhile%28%21s%5B2%5D%29%3Bif%28D%2Elength%3E1%29%7Bthrow%22Illegal%22%7Dreturn%7Br%3AA%2Ckeyword%5Fcount%3Ax%2Cvalue%3Ay%7D%7Dcatch%28H%29%7Bif%28H%3D%3D%22Illegal%22%29%7Breturn%7Br%3A0%2Ckeyword%5Fcount%3A0%2Cvalue%3Am%28C%29%7D%7Delse%7Bthrow%20H%7D%7D%7Dfunction%20g%28t%29%7Bvar%20p%3D%7Bkeyword%5Fcount%3A0%2Cr%3A0%2Cvalue%3Am%28t%29%7D%3Bvar%20r%3Dp%3Bfor%28var%20q%20in%20e%29%7Bif%28%21e%2EhasOwnProperty%28q%29%29%7Bcontinue%7Dvar%20s%3Dd%28q%2Ct%29%3Bs%2Elanguage%3Dq%3Bif%28s%2Ekeyword%5Fcount%2Bs%2Er%3Er%2Ekeyword%5Fcount%2Br%2Er%29%7Br%3Ds%7Dif%28s%2Ekeyword%5Fcount%2Bs%2Er%3Ep%2Ekeyword%5Fcount%2Bp%2Er%29%7Br%3Dp%3Bp%3Ds%7D%7Dif%28r%2Elanguage%29%7Bp%2Esecond%5Fbest%3Dr%7Dreturn%20p%7Dfunction%20i%28r%2Cq%2Cp%29%7Bif%28q%29%7Br%3Dr%2Ereplace%28%2F%5E%28%28%3C%5B%5E%3E%5D%2B%3E%7C%5Ct%29%2B%29%2Fgm%2Cfunction%28t%2Cw%2Cv%2Cu%29%7Breturn%20w%2Ereplace%28%2F%5Ct%2Fg%2Cq%29%7D%29%7Dif%28p%29%7Br%3Dr%2Ereplace%28%2F%5Cn%2Fg%2C%22%3Cbr%3E%22%29%7Dreturn%20r%7Dfunction%20n%28t%2Cw%2Cr%29%7Bvar%20x%3Dh%28t%2Cr%29%3Bvar%20v%3Da%28t%29%3Bvar%20y%2Cs%3Bif%28v%29%7By%3Dd%28v%2Cx%29%7Delse%7Breturn%7Dvar%20q%3Dc%28t%29%3Bif%28q%2Elength%29%7Bs%3Ddocument%2EcreateElement%28%22pre%22%29%3Bs%2EinnerHTML%3Dy%2Evalue%3By%2Evalue%3Dk%28q%2Cc%28s%29%2Cx%29%7Dy%2Evalue%3Di%28y%2Evalue%2Cw%2Cr%29%3Bvar%20u%3Dt%2EclassName%3Bif%28%21u%2Ematch%28%22%28%5C%5Cs%7C%5E%29%28language%2D%29%3F%22%2Bv%2B%22%28%5C%5Cs%7C%24%29%22%29%29%7Bu%3Du%3F%28u%2B%22%20%22%2Bv%29%3Av%7Dif%28%2FMSIE%20%5B678%5D%2F%2Etest%28navigator%2EuserAgent%29%26%26t%2EtagName%3D%3D%22CODE%22%26%26t%2EparentNode%2EtagName%3D%3D%22PRE%22%29%7Bs%3Dt%2EparentNode%3Bvar%20p%3Ddocument%2EcreateElement%28%22div%22%29%3Bp%2EinnerHTML%3D%22%3Cpre%3E%3Ccode%3E%22%2By%2Evalue%2B%22%3C%2Fcode%3E%3C%2Fpre%3E%22%3Bt%3Dp%2EfirstChild%2EfirstChild%3Bp%2EfirstChild%2EcN%3Ds%2EcN%3Bs%2EparentNode%2EreplaceChild%28p%2EfirstChild%2Cs%29%7Delse%7Bt%2EinnerHTML%3Dy%2Evalue%7Dt%2EclassName%3Du%3Bt%2Eresult%3D%7Blanguage%3Av%2Ckw%3Ay%2Ekeyword%5Fcount%2Cre%3Ay%2Er%7D%3Bif%28y%2Esecond%5Fbest%29%7Bt%2Esecond%5Fbest%3D%7Blanguage%3Ay%2Esecond%5Fbest%2Elanguage%2Ckw%3Ay%2Esecond%5Fbest%2Ekeyword%5Fcount%2Cre%3Ay%2Esecond%5Fbest%2Er%7D%7D%7Dfunction%20o%28%29%7Bif%28o%2Ecalled%29%7Breturn%7Do%2Ecalled%3Dtrue%3Bvar%20r%3Ddocument%2EgetElementsByTagName%28%22pre%22%29%3Bfor%28var%20p%3D0%3Bp%3Cr%2Elength%3Bp%2B%2B%29%7Bvar%20q%3Db%28r%5Bp%5D%29%3Bif%28q%29%7Bn%28q%2Chljs%2EtabReplace%29%7D%7D%7Dfunction%20l%28%29%7Bif%28window%2EaddEventListener%29%7Bwindow%2EaddEventListener%28%22DOMContentLoaded%22%2Co%2Cfalse%29%3Bwindow%2EaddEventListener%28%22load%22%2Co%2Cfalse%29%7Delse%7Bif%28window%2EattachEvent%29%7Bwindow%2EattachEvent%28%22onload%22%2Co%29%7Delse%7Bwindow%2Eonload%3Do%7D%7D%7Dvar%20e%3D%7B%7D%3Bthis%2ELANGUAGES%3De%3Bthis%2Ehighlight%3Dd%3Bthis%2EhighlightAuto%3Dg%3Bthis%2EfixMarkup%3Di%3Bthis%2EhighlightBlock%3Dn%3Bthis%2EinitHighlighting%3Do%3Bthis%2EinitHighlightingOnLoad%3Dl%3Bthis%2EIR%3D%22%5Ba%2DzA%2DZ%5D%5Ba%2DzA%2DZ0%2D9%5F%5D%2A%22%3Bthis%2EUIR%3D%22%5Ba%2DzA%2DZ%5F%5D%5Ba%2DzA%2DZ0%2D9%5F%5D%2A%22%3Bthis%2ENR%3D%22%5C%5Cb%5C%5Cd%2B%28%5C%5C%2E%5C%5Cd%2B%29%3F%22%3Bthis%2ECNR%3D%22%5C%5Cb%280%5BxX%5D%5Ba%2DfA%2DF0%2D9%5D%2B%7C%28%5C%5Cd%2B%28%5C%5C%2E%5C%5Cd%2A%29%3F%7C%5C%5C%2E%5C%5Cd%2B%29%28%5BeE%5D%5B%2D%2B%5D%3F%5C%5Cd%2B%29%3F%29%22%3Bthis%2EBNR%3D%22%5C%5Cb%280b%5B01%5D%2B%29%22%3Bthis%2ERSR%3D%22%21%7C%21%3D%7C%21%3D%3D%7C%25%7C%25%3D%7C%26%7C%26%26%7C%26%3D%7C%5C%5C%2A%7C%5C%5C%2A%3D%7C%5C%5C%2B%7C%5C%5C%2B%3D%7C%2C%7C%5C%5C%2E%7C%2D%7C%2D%3D%7C%2F%7C%2F%3D%7C%3A%7C%3B%7C%3C%7C%3C%3C%7C%3C%3C%3D%7C%3C%3D%7C%3D%7C%3D%3D%7C%3D%3D%3D%7C%3E%7C%3E%3D%7C%3E%3E%7C%3E%3E%3D%7C%3E%3E%3E%7C%3E%3E%3E%3D%7C%5C%5C%3F%7C%5C%5C%5B%7C%5C%5C%7B%7C%5C%5C%28%7C%5C%5C%5E%7C%5C%5C%5E%3D%7C%5C%5C%7C%7C%5C%5C%7C%3D%7C%5C%5C%7C%5C%5C%7C%7C%7E%22%3Bthis%2EER%3D%22%28%3F%21%5B%5C%5Cs%5C%5CS%5D%29%22%3Bthis%2EBE%3D%7Bb%3A%22%5C%5C%5C%5C%2E%22%2Cr%3A0%7D%3Bthis%2EASM%3D%7BcN%3A%22string%22%2Cb%3A%22%27%22%2Ce%3A%22%27%22%2Ci%3A%22%5C%5Cn%22%2Cc%3A%5Bthis%2EBE%5D%2Cr%3A0%7D%3Bthis%2EQSM%3D%7BcN%3A%22string%22%2Cb%3A%27%22%27%2Ce%3A%27%22%27%2Ci%3A%22%5C%5Cn%22%2Cc%3A%5Bthis%2EBE%5D%2Cr%3A0%7D%3Bthis%2ECLCM%3D%7BcN%3A%22comment%22%2Cb%3A%22%2F%2F%22%2Ce%3A%22%24%22%7D%3Bthis%2ECBLCLM%3D%7BcN%3A%22comment%22%2Cb%3A%22%2F%5C%5C%2A%22%2Ce%3A%22%5C%5C%2A%2F%22%7D%3Bthis%2EHCM%3D%7BcN%3A%22comment%22%2Cb%3A%22%23%22%2Ce%3A%22%24%22%7D%3Bthis%2ENM%3D%7BcN%3A%22number%22%2Cb%3Athis%2ENR%2Cr%3A0%7D%3Bthis%2ECNM%3D%7BcN%3A%22number%22%2Cb%3Athis%2ECNR%2Cr%3A0%7D%3Bthis%2EBNM%3D%7BcN%3A%22number%22%2Cb%3Athis%2EBNR%2Cr%3A0%7D%3Bthis%2Einherit%3Dfunction%28r%2Cs%29%7Bvar%20p%3D%7B%7D%3Bfor%28var%20q%20in%20r%29%7Bp%5Bq%5D%3Dr%5Bq%5D%7Dif%28s%29%7Bfor%28var%20q%20in%20s%29%7Bp%5Bq%5D%3Ds%5Bq%5D%7D%7Dreturn%20p%7D%7D%28%29%3Bhljs%2ELANGUAGES%2Ecpp%3Dfunction%28%29%7Bvar%20a%3D%7Bkeyword%3A%7B%22false%22%3A1%2C%22int%22%3A1%2C%22float%22%3A1%2C%22while%22%3A1%2C%22private%22%3A1%2C%22char%22%3A1%2C%22catch%22%3A1%2C%22export%22%3A1%2Cvirtual%3A1%2Coperator%3A2%2Csizeof%3A2%2Cdynamic%5Fcast%3A2%2Ctypedef%3A2%2Cconst%5Fcast%3A2%2C%22const%22%3A1%2Cstruct%3A1%2C%22for%22%3A1%2Cstatic%5Fcast%3A2%2Cunion%3A1%2Cnamespace%3A1%2Cunsigned%3A1%2C%22long%22%3A1%2C%22throw%22%3A1%2C%22volatile%22%3A2%2C%22static%22%3A1%2C%22protected%22%3A1%2Cbool%3A1%2Ctemplate%3A1%2Cmutable%3A1%2C%22if%22%3A1%2C%22public%22%3A1%2Cfriend%3A2%2C%22do%22%3A1%2C%22return%22%3A1%2C%22goto%22%3A1%2Cauto%3A1%2C%22void%22%3A2%2C%22enum%22%3A1%2C%22else%22%3A1%2C%22break%22%3A1%2C%22new%22%3A1%2Cextern%3A1%2Cusing%3A1%2C%22true%22%3A1%2C%22class%22%3A1%2Casm%3A1%2C%22case%22%3A1%2Ctypeid%3A1%2C%22short%22%3A1%2Creinterpret%5Fcast%3A2%2C%22default%22%3A1%2C%22double%22%3A1%2Cregister%3A1%2Cexplicit%3A1%2Csigned%3A1%2Ctypename%3A1%2C%22try%22%3A1%2C%22this%22%3A1%2C%22switch%22%3A1%2C%22continue%22%3A1%2Cwchar%5Ft%3A1%2Cinline%3A1%2C%22delete%22%3A1%2Calignof%3A1%2Cchar16%5Ft%3A1%2Cchar32%5Ft%3A1%2Cconstexpr%3A1%2Cdecltype%3A1%2Cnoexcept%3A1%2Cnullptr%3A1%2Cstatic%5Fassert%3A1%2Cthread%5Flocal%3A1%2Crestrict%3A1%2C%5FBool%3A1%2Ccomplex%3A1%7D%2Cbuilt%5Fin%3A%7Bstd%3A1%2Cstring%3A1%2Ccin%3A1%2Ccout%3A1%2Ccerr%3A1%2Cclog%3A1%2Cstringstream%3A1%2Cistringstream%3A1%2Costringstream%3A1%2Cauto%5Fptr%3A1%2Cdeque%3A1%2Clist%3A1%2Cqueue%3A1%2Cstack%3A1%2Cvector%3A1%2Cmap%3A1%2Cset%3A1%2Cbitset%3A1%2Cmultiset%3A1%2Cmultimap%3A1%2Cunordered%5Fset%3A1%2Cunordered%5Fmap%3A1%2Cunordered%5Fmultiset%3A1%2Cunordered%5Fmultimap%3A1%2Carray%3A1%2Cshared%5Fptr%3A1%7D%7D%3Breturn%7BdM%3A%7Bk%3Aa%2Ci%3A%22%3C%2F%22%2Cc%3A%5Bhljs%2ECLCM%2Chljs%2ECBLCLM%2Chljs%2EQSM%2C%7BcN%3A%22string%22%2Cb%3A%22%27%5C%5C%5C%5C%3F%2E%22%2Ce%3A%22%27%22%2Ci%3A%22%2E%22%7D%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5Cb%28%5C%5Cd%2B%28%5C%5C%2E%5C%5Cd%2A%29%3F%7C%5C%5C%2E%5C%5Cd%2B%29%28u%7CU%7Cl%7CL%7Cul%7CUL%7Cf%7CF%29%22%7D%2Chljs%2ECNM%2C%7BcN%3A%22preprocessor%22%2Cb%3A%22%23%22%2Ce%3A%22%24%22%7D%2C%7BcN%3A%22stl%5Fcontainer%22%2Cb%3A%22%5C%5Cb%28deque%7Clist%7Cqueue%7Cstack%7Cvector%7Cmap%7Cset%7Cbitset%7Cmultiset%7Cmultimap%7Cunordered%5Fmap%7Cunordered%5Fset%7Cunordered%5Fmultiset%7Cunordered%5Fmultimap%7Carray%29%5C%5Cs%2A%3C%22%2Ce%3A%22%3E%22%2Ck%3Aa%2Cr%3A10%2Cc%3A%5B%22self%22%5D%7D%5D%7D%7D%7D%28%29%3Bhljs%2ELANGUAGES%2Er%3D%7BdM%3A%7Bc%3A%5Bhljs%2EHCM%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5Cb0%5BxX%5D%5B0%2D9a%2DfA%2DF%5D%2B%5BLi%5D%3F%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5Cb%5C%5Cd%2B%28%3F%3A%5BeE%5D%5B%2B%5C%5C%2D%5D%3F%5C%5Cd%2A%29%3FL%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5Cb%5C%5Cd%2B%5C%5C%2E%28%3F%21%5C%5Cd%29%28%3F%3Ai%5C%5Cb%29%3F%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A1%7D%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5Cb%5C%5Cd%2B%28%3F%3A%5C%5C%2E%5C%5Cd%2A%29%3F%28%3F%3A%5BeE%5D%5B%2B%5C%5C%2D%5D%3F%5C%5Cd%2A%29%3Fi%3F%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%2C%7BcN%3A%22number%22%2Cb%3A%22%5C%5C%2E%5C%5Cd%2B%28%3F%3A%5BeE%5D%5B%2B%5C%5C%2D%5D%3F%5C%5Cd%2A%29%3Fi%3F%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A1%7D%2C%7BcN%3A%22keyword%22%2Cb%3A%22%28%3F%3AtryCatch%7Clibrary%7CsetGeneric%7CsetGroupGeneric%29%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A10%7D%2C%7BcN%3A%22keyword%22%2Cb%3A%22%5C%5C%2E%5C%5C%2E%5C%5C%2E%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A10%7D%2C%7BcN%3A%22keyword%22%2Cb%3A%22%5C%5C%2E%5C%5C%2E%5C%5Cd%2B%28%3F%21%5B%5C%5Cw%2E%5D%29%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A10%7D%2C%7BcN%3A%22keyword%22%2Cb%3A%22%5C%5Cb%28%3F%3Afunction%29%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A2%7D%2C%7BcN%3A%22keyword%22%2Cb%3A%22%28%3F%3Aif%7Cin%7Cbreak%7Cnext%7Crepeat%7Celse%7Cfor%7Creturn%7Cswitch%7Cwhile%7Ctry%7Cstop%7Cwarning%7Crequire%7Cattach%7Cdetach%7Csource%7CsetMethod%7CsetClass%29%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A1%7D%2C%7BcN%3A%22literal%22%2Cb%3A%22%28%3F%3ANA%7CNA%5Finteger%5F%7CNA%5Freal%5F%7CNA%5Fcharacter%5F%7CNA%5Fcomplex%5F%29%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A10%7D%2C%7BcN%3A%22literal%22%2Cb%3A%22%28%3F%3ANULL%7CTRUE%7CFALSE%7CT%7CF%7CInf%7CNaN%29%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A1%7D%2C%7BcN%3A%22identifier%22%2Cb%3A%22%5Ba%2DzA%2DZ%2E%5D%5Ba%2DzA%2DZ0%2D9%2E%5F%5D%2A%5C%5Cb%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%2C%7BcN%3A%22operator%22%2Cb%3A%22%3C%5C%5C%2D%28%3F%21%5C%5Cs%2A%5C%5Cd%29%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A2%7D%2C%7BcN%3A%22operator%22%2Cb%3A%22%5C%5C%2D%3E%7C%3C%5C%5C%2D%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A1%7D%2C%7BcN%3A%22operator%22%2Cb%3A%22%25%25%7C%7E%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%7D%2C%7BcN%3A%22operator%22%2Cb%3A%22%3E%3D%7C%3C%3D%7C%3D%3D%7C%21%3D%7C%5C%5C%7C%5C%5C%7C%7C%26%26%7C%3D%7C%5C%5C%2B%7C%5C%5C%2D%7C%5C%5C%2A%7C%2F%7C%5C%5C%5E%7C%3E%7C%3C%7C%21%7C%26%7C%5C%5C%7C%7C%5C%5C%24%7C%3A%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%2C%7BcN%3A%22operator%22%2Cb%3A%22%25%22%2Ce%3A%22%25%22%2Ci%3A%22%5C%5Cn%22%2Cr%3A1%7D%2C%7BcN%3A%22identifier%22%2Cb%3A%22%60%22%2Ce%3A%22%60%22%2Cr%3A0%7D%2C%7BcN%3A%22string%22%2Cb%3A%27%22%27%2Ce%3A%27%22%27%2Cc%3A%5Bhljs%2EBE%5D%2Cr%3A0%7D%2C%7BcN%3A%22string%22%2Cb%3A%22%27%22%2Ce%3A%22%27%22%2Cc%3A%5Bhljs%2EBE%5D%2Cr%3A0%7D%2C%7BcN%3A%22paren%22%2Cb%3A%22%5B%5B%28%7B%5C%5C%5D%29%7D%5D%22%2Ce%3Ahljs%2EIMMEDIATE%5FRE%2Cr%3A0%7D%5D%7D%7D%3B%0Ahljs%2EinitHighlightingOnLoad%28%29%3B%0A%0A"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link href="data:text/css,body%2C%20td%20%7B%0A%20%20%20font%2Dfamily%3A%20sans%2Dserif%3B%0A%20%20%20background%2Dcolor%3A%20white%3B%0A%20%20%20font%2Dsize%3A%2013px%3B%0A%7D%0A%0Abody%20%7B%0A%20%20max%2Dwidth%3A%20800px%3B%0A%20%20margin%3A%200%20auto%3B%0A%20%20padding%3A%201em%201em%202em%3B%0A%20%20line%2Dheight%3A%2020px%3B%0A%7D%0A%0A%2F%2A%20Table%20of%20contents%20style%20%2A%2F%0A%0Adiv%23TOC%20li%20%7B%0A%20%20%20%20list%2Dstyle%3Anone%3B%0A%20%20%20%20background%2Dimage%3Anone%3B%0A%20%20%20%20background%2Drepeat%3Anone%3B%0A%20%20%20%20background%2Dposition%3A0%3B%0A%7D%0A%0A%2F%2A%20element%20spacing%20%2A%2F%0A%0Ap%2C%20pre%20%7B%20%0A%20%20margin%3A%200em%200em%201em%3B%0A%7D%0A%0A%2F%2A%20center%20images%20and%20tables%20%2A%2F%0Aimg%2C%20table%20%7B%0A%20%20margin%3A%200em%20auto%201em%3B%0A%7D%0A%0Ap%20%7B%0A%20%20text%2Dalign%3A%20justify%3B%0A%7D%0A%0Att%2C%20code%2C%20pre%20%7B%0A%20%20%20font%2Dfamily%3A%20%27DejaVu%20Sans%20Mono%27%2C%20%27Droid%20Sans%20Mono%27%2C%20%27Lucida%20Console%27%2C%20Consolas%2C%20Monaco%2C%20monospace%3B%0A%7D%0A%0Ah1%2C%20h2%2C%20h3%2C%20h4%2C%20h5%2C%20h6%20%7B%20%0A%20%20font%2Dfamily%3A%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20margin%3A%201%2E2em%200em%200%2E6em%200em%3B%0A%20%20font%2Dweight%3A%20bold%3B%0A%7D%0A%0Ah1%2Etitle%20%7B%0A%20%20font%2Dsize%3A%20250%25%3B%0A%20%20font%2Dweight%3A%20normal%3B%0A%20%20color%3A%20%2387b13f%3B%0A%20%20line%2Dheight%3A%201%2E1em%3B%0A%20%20margin%2Dtop%3A%200px%3B%0A%20%20border%2Dbottom%3A%200px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20font%2Dsize%3A%20160%25%3B%0A%20%20font%2Dweight%3A%20normal%3B%0A%20%20line%2Dheight%3A%201%2E4em%3B%0A%20%20border%2Dbottom%3A%201px%20%231a81c2%20solid%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20font%2Dsize%3A%20130%25%3B%20%20%0A%7D%0A%0Ah1%2C%20h2%2C%20h3%20%7B%0A%20%20color%3A%20%231a81c2%3B%0A%7D%0A%0Ah3%2C%20h4%2C%20h5%2C%20h6%20%7B%0A%20%20font%2Dsize%3A115%25%3B%0A%7D%20%2F%2A%20not%20expecting%20to%20dive%20deeper%20than%20four%20levels%20on%20a%20single%20page%20%2A%2F%0A%0A%2F%2A%20links%20are%20simply%20blue%2C%20hovering%20slightly%20less%20blue%20%2A%2F%0Aa%20%7B%20color%3A%20%231a81c2%3B%20%7D%0Aa%3Aactive%20%7B%20outline%3A%20none%3B%20%7D%0Aa%3Avisited%20%7B%20color%3A%20%231a81c2%3B%20%7D%0Aa%3Ahover%20%7B%20color%3A%20%234c94c2%3B%20%7D%0A%0Apre%2C%20img%20%7B%0A%20%20max%2Dwidth%3A%20100%25%3B%0A%20%20display%3A%20block%3B%0A%7D%0A%0Apre%20%7B%0A%20%20border%3A%200px%20none%3B%0A%20%20background%2Dcolor%3A%20%23F8F8F8%3B%0A%20%20white%2Dspace%3A%20pre%3B%0A%20%20overflow%2Dx%3A%20auto%3B%0A%7D%0A%0Apre%20code%20%7B%0A%20%20border%3A%201px%20%23aaa%20dashed%3B%0A%20%20background%2Dcolor%3A%20white%3B%0A%20%20display%3A%20block%3B%0A%20%20padding%3A%201em%3B%20%20%0A%20%20color%3A%20%23111%3B%0A%20%20overflow%2Dx%3A%20inherit%3B%0A%7D%0A%0A%2F%2A%20markdown%20v1%20%2A%2F%0Apre%20code%5Bclass%5D%20%7B%0A%20%20background%2Dcolor%3A%20inherit%3B%0A%7D%0A%0A%2F%2A%20markdown%20v2%20%2A%2F%0Apre%5Bclass%5D%20code%20%7B%0A%20%20background%2Dcolor%3A%20inherit%3B%0A%7D%0A%0A%2F%2A%20formatting%20of%20inline%20code%20%2A%2F%0Acode%20%7B%20%0A%20%20background%2Dcolor%3A%20transparent%3B%0A%20%20color%3A%20%2387b13f%3B%0A%20%20font%2Dsize%3A%2092%25%3B%0A%7D%0A%0A%2F%2A%20formatting%20of%20tables%20%2A%2F%0A%0Atable%2C%20td%2C%20th%20%7B%0A%20%20border%3A%20none%3B%0A%20%20padding%3A%200%200%2E5em%3B%0A%7D%0A%0A%2F%2A%20alternating%20row%20colors%20%2A%2F%0Atbody%20tr%3Anth%2Dchild%28odd%29%20td%20%7B%0A%20%20background%2Dcolor%3A%20%23F8F8F8%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20%20color%3A%23666666%3B%0A%20%20%20margin%3A0%3B%0A%20%20%20padding%2Dleft%3A%201em%3B%0A%20%20%20border%2Dleft%3A%200%2E5em%20%23EEE%20solid%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20%20height%3A%200px%3B%0A%20%20%20border%2Dbottom%3A%20none%3B%0A%20%20%20border%2Dtop%2Dwidth%3A%20thin%3B%0A%20%20%20border%2Dtop%2Dstyle%3A%20dotted%3B%0A%20%20%20border%2Dtop%2Dcolor%3A%20%23999999%3B%0A%7D%0A%0Aspan%2Eheader%2Dsection%2Dnumber%20%7B%0A%20%20padding%2Dright%3A%201em%3B%0A%7D%0A%0Aspan%2Etoc%2Dsection%2Dnumber%3A%3Aafter%20%7B%0A%20%20%20%20content%3A%20%22%20%20%22%3B%0A%20%20%20%20white%2Dspace%3A%20pre%3B%0A%7D%0A%0A%40media%20print%20%7B%0A%20%20%20%2A%20%7B%0A%20%20%20%20%20%20background%3A%20transparent%20%21important%3B%0A%20%20%20%20%20%20color%3A%20black%20%21important%3B%0A%20%20%20%20%20%20filter%3Anone%20%21important%3B%0A%20%20%20%20%20%20%2Dms%2Dfilter%3A%20none%20%21important%3B%0A%20%20%20%7D%0A%0A%20%20%20body%20%7B%0A%20%20%20%20%20%20font%2Dsize%3A12pt%3B%0A%20%20%20%20%20%20max%2Dwidth%3A100%25%3B%0A%20%20%20%7D%0A%0A%20%20%20a%2C%20a%3Avisited%20%7B%0A%20%20%20%20%20%20text%2Ddecoration%3A%20underline%3B%0A%20%20%20%7D%0A%0A%20%20%20hr%20%7B%0A%20%20%20%20%20%20visibility%3A%20hidden%3B%0A%20%20%20%20%20%20page%2Dbreak%2Dbefore%3A%20always%3B%0A%20%20%20%7D%0A%0A%20%20%20pre%2C%20blockquote%20%7B%0A%20%20%20%20%20%20padding%2Dright%3A%201em%3B%0A%20%20%20%20%20%20page%2Dbreak%2Dinside%3A%20avoid%3B%0A%20%20%20%7D%0A%0A%20%20%20tr%2C%20img%20%7B%0A%20%20%20%20%20%20page%2Dbreak%2Dinside%3A%20avoid%3B%0A%20%20%20%7D%0A%0A%20%20%20img%20%7B%0A%20%20%20%20%20%20max%2Dwidth%3A%20100%25%20%21important%3B%0A%20%20%20%7D%0A%0A%20%20%20%40page%20%3Aleft%20%7B%0A%20%20%20%20%20%20margin%3A%2015mm%2020mm%2015mm%2010mm%3B%0A%20%20%20%7D%0A%0A%20%20%20%40page%20%3Aright%20%7B%0A%20%20%20%20%20%20margin%3A%2015mm%2010mm%2015mm%2020mm%3B%0A%20%20%20%7D%0A%0A%20%20%20p%2C%20h2%2C%20h3%20%7B%0A%20%20%20%20%20%20orphans%3A%203%3B%20widows%3A%203%3B%0A%20%20%20%7D%0A%0A%20%20%20h2%2C%20h3%20%7B%0A%20%20%20%20%20%20page%2Dbreak%2Dafter%3A%20avoid%3B%0A%20%20%20%7D%0A%7D%0A" rel="stylesheet" type="text/css" />

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if(links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

</head>

<body>


<div id="header">
<h1 class="title">Example report using bumphunter results</h1>
<h4 class="author"><em>L Collado-Torres</em></h4>
<h4 class="date"><em>1 April 2016</em></h4>
</div>
<h4 class="package">Package version: <span style="font-weight: normal">regionReport 1.5.30</span></h4>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#bumphunter-example"><span class="toc-section-number">1</span> <em><a href="http://bioconductor.org/packages/bumphunter">bumphunter</a></em> example</a></li>
<li><a href="#reproducibility"><span class="toc-section-number">2</span> Reproducibility</a></li>
</ul>
</div>

<div id="bumphunter-example" class="section level1">
<h1><span class="header-section-number">1</span> <em><a href="http://bioconductor.org/packages/bumphunter">bumphunter</a></em> example</h1>
<p>The <em><a href="http://bioconductor.org/packages/bumphunter">bumphunter</a></em> package can be used for methylation analyses where you are interested in identifying differentially methylated regions. The <a href="http://bioconductor.org/packages/release/bioc/vignettes/bumphunter/inst/doc/bumphunter.pdf">vignette</a> explains in greater detail the data set we are using in this example.</p>
<pre class="r"><code>## Load bumphunter
library('bumphunter')</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, cbind, colnames,
##     do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, lengths, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rownames, sapply, setdiff,
##     sort, table, tapply, union, unique, unsplit</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     colMeans, colSums, expand.grid, rowMeans, rowSums</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: foreach</code></pre>
<pre><code>## foreach: simple, scalable parallel programming from Revolution Analytics
## Use Revolution R for scalability, fault tolerance and more.
## http://www.revolutionanalytics.com</code></pre>
<pre><code>## Loading required package: iterators</code></pre>
<pre><code>## Loading required package: locfit</code></pre>
<pre><code>## locfit 1.5-9.1    2013-03-22</code></pre>
<pre class="r"><code>## Create data from the vignette
pos &lt;- list(pos1=seq(1, 1000, 35),
            pos2=seq(2001, 3000, 35),
            pos3=seq(1, 1000, 50))
chr &lt;- rep(paste0('chr', c(1, 1, 2)), times = sapply(pos, length))
pos &lt;- unlist(pos, use.names = FALSE)

## Find clusters
cl &lt;- clusterMaker(chr, pos, maxGap = 300)

## Build simulated bumps
Indexes &lt;- split(seq_along(cl), cl)
beta1 &lt;- rep(0, length(pos))
for(i in seq(along=Indexes)){
    ind &lt;- Indexes[[i]]
    x &lt;- pos[ind]
    z &lt;- scale(x, median(x), max(x)/12)
    beta1[ind] &lt;- i*(-1)^(i+1)*pmax(1-abs(z)^3,0)^3 ##multiply by i to vary size
}

## Build data
beta0 &lt;- 3 * sin(2 * pi * pos / 720)
X &lt;- cbind(rep(1, 20), rep(c(0, 1), each = 10))
set.seed(23852577)
error &lt;- matrix(rnorm(20 * length(beta1), 0, 1), ncol = 20)
y &lt;- t(X[, 1]) %x% beta0 + t(X[, 2]) %x% beta1 + error

## Perform bumphunting
tab &lt;- bumphunter(y, X, chr, pos, cl, cutoff=.5)</code></pre>
<pre><code>## [bumphunterEngine] Using a single core (backend: doSEQ, version: 1.4.3).</code></pre>
<pre><code>## [bumphunterEngine] Computing coefficients.</code></pre>
<pre><code>## [bumphunterEngine] Finding regions.</code></pre>
<pre><code>## [bumphunterEngine] Found 15 bumps.</code></pre>
<pre class="r"><code>## Explore data
lapply(tab, head)</code></pre>
<pre><code>## $table
##     chr start  end      value       area cluster indexStart indexEnd  L
## 10 chr1  2316 2631 -1.5814747 15.8147473       2         39       48 10
## 7  chr2   451  551  1.5891293  4.7673878       3         68       70  3
## 2  chr1   456  526  1.0678828  3.2036485       1         14       16  3
## 5  chr1  2176 2211  0.7841794  1.5683589       2         35       36  2
## 6  chr1  2841 2841  1.2010184  1.2010184       2         54       54  1
## 4  chr1   771  771  0.7780902  0.7780902       1         23       23  1
##    clusterL
## 10       29
## 7        20
## 2        29
## 5        29
## 6        29
## 4        29
## 
## $coef
##             [,1]
## [1,]  0.60960932
## [2,] -0.09052769
## [3,] -0.21482638
## [4,]  0.13053755
## [5,] -0.21723642
## [6,]  0.39934961
## 
## $fitted
##             [,1]
## [1,]  0.60960932
## [2,] -0.09052769
## [3,] -0.21482638
## [4,]  0.13053755
## [5,] -0.21723642
## [6,]  0.39934961
## 
## $pvaluesMarginal
## [1] NA</code></pre>
<p>Once we have the regions we can proceed to build the required <code>GRanges</code> object.</p>
<pre class="r"><code>library('GenomicRanges')

## Build GRanges with sequence lengths
regions &lt;- GRanges(seqnames = tab$table$chr, 
    IRanges(start = tab$table$start, end = tab$table$end),
    strand = '*', value = tab$table$value, area = tab$table$area, 
    cluster = tab$table$cluster, L = tab$table$L, clusterL = tab$table$clusterL)

## Assign chr lengths
data(hg19Ideogram, package = 'biovizBase')
seqlengths(regions) &lt;- seqlengths(hg19Ideogram)[names(seqlengths(regions))]

## Explore the regions
regions</code></pre>
<pre><code>## GRanges object with 15 ranges and 5 metadata columns:
##        seqnames       ranges strand |              value              area
##           &lt;Rle&gt;    &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;numeric&gt;         &lt;numeric&gt;
##    [1]     chr1 [2316, 2631]      * |  -1.58147472668612  15.8147472668612
##    [2]     chr2 [ 451,  551]      * |   1.58912926544627  4.76738779633882
##    [3]     chr1 [ 456,  526]      * |   1.06788283103948  3.20364849311844
##    [4]     chr1 [2176, 2211]      * |  0.784179442360784  1.56835888472157
##    [5]     chr1 [2841, 2841]      * |   1.20101842543356  1.20101842543356
##    ...      ...          ...    ... .                ...               ...
##   [11]     chr1 [ 631,  631]      * |  0.618602646982714 0.618602646982714
##   [12]     chr1 [   1,    1]      * |  0.609609318450113 0.609609318450113
##   [13]     chr1 [2911, 2911]      * | -0.576423040796044 0.576423040796044
##   [14]     chr2 [ 251,  251]      * | -0.556159535339166 0.556159535339166
##   [15]     chr1 [2806, 2806]      * | -0.521605772372843 0.521605772372843
##          cluster         L  clusterL
##        &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##    [1]         2        10        29
##    [2]         3         3        20
##    [3]         1         3        29
##    [4]         2         2        29
##    [5]         2         1        29
##    ...       ...       ...       ...
##   [11]         1         1        29
##   [12]         1         1        29
##   [13]         2         1        29
##   [14]         3         1        20
##   [15]         2         1        29
##   -------
##   seqinfo: 2 sequences from an unspecified genome</code></pre>
<p>Now that we have identified a set of differentially methylated regions we can proceed to creating the HTML report. Note that this report has less information than the <a href="http://leekgroup.github.io/regionReportSupp/DiffBind.html">DiffBind example</a> because we don’t have a p-value variable.</p>
<pre class="r"><code>## Load regionReport
library('regionReport')</code></pre>
<pre class="r"><code>## Make it so that the report will be available as a vignette
original &lt;- readLines(system.file('regionExploration', 'regionExploration.Rmd',
    package = 'regionReport'))
vignetteInfo &lt;- c(
    'vignette: &gt;',
    '  %\\VignetteEngine{knitr::rmarkdown}',
    '  %\\VignetteIndexEntry{Basic genomic regions exploration}',
    '  %\\VignetteEncoding{UTF-8}'
)
new &lt;- c(original[1:12], vignetteInfo, original[13:length(original)])
writeLines(new, '/Users/lcollado/Dropbox/JHSPH/Code/regionReportSupp/bumphunter-example/regionReportBumphunter.Rmd')

## Now create the report
report &lt;- renderReport(regions, 'Example bumphunter', pvalueVars = NULL,
    densityVars = c('Area' = 'area', 'Value' = 'value',
    'Cluster Length' = 'clusterL'), significantVar = NULL,
    output = 'index', outdir = 'bumphunter-example',
    template = '/Users/lcollado/Dropbox/JHSPH/Code/regionReportSupp/bumphunter-example/regionReportBumphunter.Rmd', device = 'png')</code></pre>
<pre><code>## Writing 11 Bibtex entries ...</code></pre>
<pre><code>## OK
## Results written to file 'bumphunter-example/index.bib'</code></pre>
<pre><code>## 
## 
## processing file: index.Rmd</code></pre>
<pre><code>## 
  |                                                                       
  |                                                                 |   0%
  |                                                                       
  |..                                                               |   2%
##    inline R code fragments
## 
## 
  |                                                                       
  |...                                                              |   5%
## label: setup (with options) 
## List of 1
##  $ bootstrap.show.message: logi FALSE
## 
## 
  |                                                                       
  |.....                                                            |   7%
##   ordinary text without R code
## 
## 
  |                                                                       
  |......                                                           |  10%
## label: pvaluePlots (with options) 
## List of 3
##  $ echo   : logi FALSE
##  $ results: chr &quot;asis&quot;
##  $ eval   : symbol hasPvalueVars
## 
## 
  |                                                                       
  |........                                                         |  12%
##   ordinary text without R code
## 
## 
  |                                                                       
  |..........                                                       |  15%
## label: regLen (with options) 
## List of 5
##  $ fig.width : num 14
##  $ fig.height: num 14
##  $ dev       : symbol device
##  $ eval      : symbol hasSignificant
##  $ echo      : symbol hasSignificant
## 
## 
  |                                                                       
  |...........                                                      |  17%
##   ordinary text without R code
## 
## 
  |                                                                       
  |.............                                                    |  20%
## label: regLen2 (with options) 
## List of 5
##  $ fig.width : num 10
##  $ fig.height: num 10
##  $ dev       : symbol device
##  $ eval      : language !hasSignificant
##  $ echo      : language !hasSignificant</code></pre>
<pre><code>## 
  |                                                                       
  |..............                                                   |  22%
##    inline R code fragments
## 
## 
  |                                                                       
  |................                                                 |  24%
## label: densityPlots (with options) 
## List of 3
##  $ echo   : logi FALSE
##  $ results: chr &quot;asis&quot;
##  $ eval   : symbol hasDensityVars</code></pre>
<pre><code>## 
  |                                                                       
  |.................                                                |  27%
##    inline R code fragments
## 
## 
  |                                                                       
  |...................                                              |  29%
## label: genomeOverview1 (with options) 
## List of 7
##  $ message   : logi FALSE
##  $ fig.width : num 7
##  $ fig.height: num 9
##  $ dev       : symbol device
##  $ dpi       : num 300
##  $ eval      : symbol hasSignificant
##  $ echo      : symbol hasSignificant
## 
## 
  |                                                                       
  |.....................                                            |  32%
##    inline R code fragments
## 
## 
  |                                                                       
  |......................                                           |  34%
## label: manhattanPlots (with options) 
## List of 3
##  $ echo   : logi FALSE
##  $ results: chr &quot;asis&quot;
##  $ eval   : symbol hasPvalueVars
## 
## 
  |                                                                       
  |........................                                         |  37%
##   ordinary text without R code
## 
## 
  |                                                                       
  |.........................                                        |  39%
## label: genomeOverview2 (with options) 
## List of 5
##  $ message   : logi FALSE
##  $ fig.width : num 7
##  $ fig.height: num 9
##  $ dev       : symbol device
##  $ dpi       : num 300</code></pre>
<pre><code>## No annotationPackage supplied. Trying org.Hs.eg.db.</code></pre>
<pre><code>## Loading required package: org.Hs.eg.db</code></pre>
<pre><code>## Loading required package: DBI</code></pre>
<pre><code>## </code></pre>
<pre><code>## Getting TSS and TSE.</code></pre>
<pre><code>## Getting CSS and CSE.</code></pre>
<pre><code>## Getting exons.</code></pre>
<pre><code>## Annotating genes.</code></pre>
<pre><code>## Scale for 'x' is already present. Adding another scale for 'x', which
## will replace the existing scale.</code></pre>
<pre><code>## 
  |                                                                       
  |...........................                                      |  41%
##    inline R code fragments
## 
## 
  |                                                                       
  |.............................                                    |  44%
## label: annoReg (with options) 
## List of 1
##  $ results: chr &quot;asis&quot;
## 
## 
  |                                                                       
  |..............................                                   |  46%
##   ordinary text without R code
## 
## 
  |                                                                       
  |................................                                 |  49%
## label: genomeOverview3 (with options) 
## List of 7
##  $ message   : logi FALSE
##  $ fig.width : num 7
##  $ fig.height: num 9
##  $ dev       : symbol device
##  $ dpi       : num 300
##  $ eval      : symbol hasSignificant
##  $ echo      : symbol hasSignificant
## 
## 
  |                                                                       
  |.................................                                |  51%
##    inline R code fragments
## 
## 
  |                                                                       
  |...................................                              |  54%
## label: countTable (with options) 
## List of 1
##  $ results: chr &quot;asis&quot;
## 
## 
  |                                                                       
  |....................................                             |  56%
##   ordinary text without R code
## 
## 
  |                                                                       
  |......................................                           |  59%
## label: vennDiagram (with options) 
## List of 3
##  $ dev       : symbol device
##  $ fig.width : num 7
##  $ fig.height: num 7</code></pre>
<pre><code>## 
  |                                                                       
  |........................................                         |  61%
##    inline R code fragments
## 
## 
  |                                                                       
  |.........................................                        |  63%
## label: vennDiagramSignificant (with options) 
## List of 5
##  $ eval      : symbol hasSignificant
##  $ echo      : symbol hasSignificant
##  $ dev       : symbol device
##  $ fig.width : num 7
##  $ fig.height: num 7
## 
## 
  |                                                                       
  |...........................................                      |  66%
##    inline R code fragments
## 
## 
  |                                                                       
  |............................................                     |  68%
## label: bestRegionInfo (with options) 
## List of 1
##  $ results: chr &quot;asis&quot;
## 
## 
  |                                                                       
  |..............................................                   |  71%
##   ordinary text without R code
## 
## 
  |                                                                       
  |................................................                 |  73%
## label: unnamed-chunk-1 (with options) 
## List of 3
##  $ child: symbol customCode
##  $ eval : symbol hasCustomCode
##  $ dev  : symbol device
## 
## 
  |                                                                       
  |.................................................                |  76%
##    inline R code fragments
## 
## 
  |                                                                       
  |...................................................              |  78%
## label: thecall (with options) 
## List of 1
##  $ echo: logi FALSE
## 
## 
  |                                                                       
  |....................................................             |  80%
##   ordinary text without R code
## 
## 
  |                                                                       
  |......................................................           |  83%
## label: reproducibility1 (with options) 
## List of 1
##  $ echo: logi FALSE
## 
## 
  |                                                                       
  |.......................................................          |  85%
##   ordinary text without R code
## 
## 
  |                                                                       
  |.........................................................        |  88%
## label: reproducibility2 (with options) 
## List of 1
##  $ echo: logi FALSE
## 
## 
  |                                                                       
  |...........................................................      |  90%
##   ordinary text without R code
## 
## 
  |                                                                       
  |............................................................     |  93%
## label: reproducibility3 (with options) 
## List of 1
##  $ echo: logi FALSE
## 
## 
  |                                                                       
  |..............................................................   |  95%
##    inline R code fragments
## 
## 
  |                                                                       
  |...............................................................  |  98%
## label: bibliography (with options) 
## List of 3
##  $ results: chr &quot;asis&quot;
##  $ echo   : logi FALSE
##  $ warning: logi FALSE
## 
## 
  |                                                                       
  |.................................................................| 100%
##   ordinary text without R code</code></pre>
<pre><code>## output file: index.knit.md</code></pre>
<pre><code>## /usr/local/bin/pandoc +RTS -K512m -RTS index.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash-implicit_figures --output index.html --smart --email-obfuscation none --self-contained --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --template /Library/Frameworks/R.framework/Versions/3.3/Resources/library/rmarkdown/rmd/h/default.html --variable 'theme:spacelab' --include-in-header /var/folders/cx/n9s558kx6fb7jf5z_pgszgb80000gn/T//RtmpABKBrT/rmarkdown-str9af175e4d8ac.html --mathjax --variable 'mathjax-url:https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --no-highlight --variable highlightjs=/Library/Frameworks/R.framework/Versions/3.3/Resources/library/rmarkdown/rmd/h/highlight --variable navigationjs=/Library/Frameworks/R.framework/Versions/3.3/Resources/library/rmarkdown/rmd/h/navigation-1.0 --variable code_folding=hide</code></pre>
<pre><code>## 
## Output created: index.html</code></pre>
<pre class="r"><code>## Clean up
file.remove('/Users/lcollado/Dropbox/JHSPH/Code/regionReportSupp/bumphunter-example/regionReportBumphunter.Rmd')</code></pre>
<pre><code>## [1] TRUE</code></pre>
<p>You can view the final report <a href="/regionReportSupp/bumphunter-example/index.html">here</a>.</p>
</div>
<div id="reproducibility" class="section level1">
<h1><span class="header-section-number">2</span> Reproducibility</h1>
<pre class="r"><code>## Date generated:
Sys.time()</code></pre>
<pre><code>## [1] &quot;2016-04-01 05:01:22 EDT&quot;</code></pre>
<pre class="r"><code>## Time spent making this page:
proc.time()</code></pre>
<pre><code>##    user  system elapsed 
## 130.696   5.251 136.121</code></pre>
<pre class="r"><code>## R and packages info:
options(width = 120)
devtools::session_info()</code></pre>
<pre><code>## Session info -----------------------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                                    
##  version  R version 3.3.0 alpha (2016-03-23 r70368)
##  system   x86_64, darwin13.4.0                     
##  ui       AQUA                                     
##  language (EN)                                     
##  collate  en_US.UTF-8                              
##  tz       America/New_York                         
##  date     2016-04-01</code></pre>
<pre><code>## Packages ---------------------------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package                           * version  date       source                                   
##  acepack                             1.3-3.3  2014-11-24 CRAN (R 3.3.0)                           
##  annotate                            1.49.1   2016-02-06 Bioconductor                             
##  AnnotationDbi                     * 1.33.7   2016-01-29 Bioconductor                             
##  AnnotationHub                       2.3.16   2016-03-25 Bioconductor                             
##  bibtex                              0.4.0    2014-12-31 CRAN (R 3.3.0)                           
##  Biobase                           * 2.31.3   2016-01-14 Bioconductor                             
##  BiocGenerics                      * 0.17.3   2016-01-29 Bioconductor                             
##  BiocInstaller                       1.21.4   2016-03-23 Bioconductor                             
##  BiocParallel                        1.5.21   2016-03-23 Bioconductor                             
##  BiocStyle                         * 1.9.8    2016-03-25 Bioconductor                             
##  biomaRt                             2.27.2   2016-01-14 Bioconductor                             
##  Biostrings                          2.39.12  2016-02-21 Bioconductor                             
##  biovizBase                          1.19.4   2016-02-22 Bioconductor                             
##  bitops                              1.0-6    2013-08-17 CRAN (R 3.3.0)                           
##  BSgenome                            1.39.4   2016-02-21 Bioconductor                             
##  bumphunter                        * 1.11.5   2016-03-29 Bioconductor                             
##  cluster                             2.0.3    2015-07-21 CRAN (R 3.3.0)                           
##  codetools                           0.2-14   2015-07-15 CRAN (R 3.3.0)                           
##  colorspace                          1.2-6    2015-03-11 CRAN (R 3.3.0)                           
##  DBI                               * 0.3.1    2014-09-24 CRAN (R 3.3.0)                           
##  derfinder                         * 1.5.30   2016-03-25 Bioconductor                             
##  derfinderHelper                     1.5.3    2016-03-23 Bioconductor                             
##  derfinderPlot                     * 1.5.7    2016-03-23 Bioconductor                             
##  DESeq2                              1.11.36  2016-03-31 Bioconductor                             
##  devtools                          * 1.10.0   2016-01-23 CRAN (R 3.3.0)                           
##  dichromat                           2.0-0    2013-01-24 CRAN (R 3.3.0)                           
##  digest                              0.6.9    2016-01-08 CRAN (R 3.3.0)                           
##  doRNG                               1.6      2014-03-07 CRAN (R 3.3.0)                           
##  DT                                * 0.1      2015-06-09 CRAN (R 3.3.0)                           
##  ensembldb                           1.3.18   2016-03-10 Bioconductor                             
##  evaluate                            0.8.3    2016-03-05 CRAN (R 3.3.0)                           
##  foreach                           * 1.4.3    2015-10-13 CRAN (R 3.3.0)                           
##  foreign                             0.8-66   2015-08-19 CRAN (R 3.3.0)                           
##  formatR                             1.3      2016-03-05 CRAN (R 3.3.0)                           
##  Formula                             1.2-1    2015-04-07 CRAN (R 3.3.0)                           
##  genefilter                          1.53.3   2016-03-23 Bioconductor                             
##  geneplotter                         1.49.0   2016-01-14 Bioconductor                             
##  GenomeInfoDb                      * 1.7.6    2016-01-29 Bioconductor                             
##  GenomicAlignments                   1.7.20   2016-02-25 Bioconductor                             
##  GenomicFeatures                   * 1.23.27  2016-03-25 Bioconductor                             
##  GenomicFiles                        1.7.9    2016-02-22 Bioconductor                             
##  GenomicRanges                     * 1.23.25  2016-03-31 Bioconductor                             
##  GGally                              1.0.1    2016-01-14 CRAN (R 3.3.0)                           
##  ggbio                             * 1.19.10  2016-02-23 Bioconductor                             
##  ggplot2                           * 2.1.0    2016-03-01 CRAN (R 3.3.0)                           
##  graph                               1.49.1   2016-01-14 Bioconductor                             
##  gridExtra                         * 2.2.1    2016-02-29 CRAN (R 3.3.0)                           
##  gtable                              0.2.0    2016-02-26 CRAN (R 3.3.0)                           
##  highr                               0.5.1    2015-09-18 CRAN (R 3.3.0)                           
##  Hmisc                               3.17-2   2016-02-21 CRAN (R 3.3.0)                           
##  htmltools                           0.3.5    2016-03-21 CRAN (R 3.3.0)                           
##  htmlwidgets                         0.6      2016-02-25 CRAN (R 3.3.0)                           
##  httpuv                              1.3.3    2015-08-04 CRAN (R 3.3.0)                           
##  httr                                1.1.0    2016-01-28 CRAN (R 3.3.0)                           
##  interactiveDisplayBase              1.9.0    2016-01-14 Bioconductor                             
##  IRanges                           * 2.5.40   2016-03-11 Bioconductor                             
##  iterators                         * 1.0.8    2015-10-13 CRAN (R 3.3.0)                           
##  jsonlite                            0.9.19   2015-11-28 CRAN (R 3.3.0)                           
##  knitcitations                       1.0.7    2015-10-28 CRAN (R 3.3.0)                           
##  knitr                             * 1.12.3   2016-01-22 CRAN (R 3.3.0)                           
##  knitrBootstrap                      1.0.0    2016-03-24 Github (jimhester/knitrBootstrap@cdaa4a9)
##  labeling                            0.3      2014-08-23 CRAN (R 3.3.0)                           
##  lattice                             0.20-33  2015-07-14 CRAN (R 3.3.0)                           
##  latticeExtra                        0.6-28   2016-02-09 CRAN (R 3.3.0)                           
##  limma                               3.27.14  2016-03-23 Bioconductor                             
##  locfit                            * 1.5-9.1  2013-04-20 CRAN (R 3.3.0)                           
##  lubridate                           1.5.0    2015-12-03 CRAN (R 3.3.0)                           
##  magrittr                            1.5      2014-11-22 CRAN (R 3.3.0)                           
##  markdown                            0.7.7    2015-04-22 CRAN (R 3.3.0)                           
##  Matrix                              1.2-4    2016-03-02 CRAN (R 3.3.0)                           
##  matrixStats                         0.50.1   2015-12-15 CRAN (R 3.3.0)                           
##  memoise                             1.0.0    2016-01-29 CRAN (R 3.3.0)                           
##  mgcv                              * 1.8-12   2016-03-03 CRAN (R 3.3.0)                           
##  mime                                0.4      2015-09-03 CRAN (R 3.3.0)                           
##  munsell                             0.4.3    2016-02-13 CRAN (R 3.3.0)                           
##  nlme                              * 3.1-126  2016-03-14 CRAN (R 3.3.0)                           
##  nnet                                7.3-12   2016-02-02 CRAN (R 3.3.0)                           
##  org.Hs.eg.db                      * 3.2.3    2016-03-24 Bioconductor                             
##  OrganismDbi                         1.13.5   2016-02-24 Bioconductor                             
##  pkgmaker                            0.22     2014-05-14 CRAN (R 3.3.0)                           
##  plyr                                1.8.3    2015-06-12 CRAN (R 3.3.0)                           
##  qvalue                              2.3.2    2016-01-14 Bioconductor                             
##  R6                                  2.1.2    2016-01-26 CRAN (R 3.3.0)                           
##  RBGL                                1.47.0   2016-01-14 Bioconductor                             
##  RColorBrewer                      * 1.1-2    2014-12-07 CRAN (R 3.3.0)                           
##  Rcpp                                0.12.4   2016-03-26 CRAN (R 3.3.0)                           
##  RCurl                               1.95-4.8 2016-03-01 CRAN (R 3.3.0)                           
##  RefManageR                          0.10.12  2016-03-31 CRAN (R 3.3.0)                           
##  regionReport                      * 1.5.30   2016-04-01 Bioconductor                             
##  registry                            0.3      2015-07-08 CRAN (R 3.3.0)                           
##  reshape                             0.8.5    2014-04-23 CRAN (R 3.3.0)                           
##  reshape2                            1.4.1    2014-12-06 CRAN (R 3.3.0)                           
##  RJSONIO                             1.3-0    2014-07-28 CRAN (R 3.3.0)                           
##  rmarkdown                           0.9.5    2016-02-22 CRAN (R 3.3.0)                           
##  rngtools                            1.2.4    2014-03-06 CRAN (R 3.3.0)                           
##  rpart                               4.1-10   2015-06-29 CRAN (R 3.3.0)                           
##  Rsamtools                           1.23.6   2016-03-23 Bioconductor                             
##  RSQLite                           * 1.0.0    2014-10-25 CRAN (R 3.3.0)                           
##  rstudioapi                          0.5      2016-01-24 CRAN (R 3.3.0)                           
##  rtracklayer                         1.31.9   2016-03-29 Bioconductor                             
##  S4Vectors                         * 0.9.44   2016-03-28 Bioconductor                             
##  scales                              0.4.0    2016-02-26 CRAN (R 3.3.0)                           
##  shiny                               0.13.2   2016-03-28 CRAN (R 3.3.0)                           
##  stringi                             1.0-1    2015-10-22 CRAN (R 3.3.0)                           
##  stringr                             1.0.0    2015-04-30 CRAN (R 3.3.0)                           
##  SummarizedExperiment                1.1.22   2016-03-10 Bioconductor                             
##  survival                            2.38-3   2015-07-02 CRAN (R 3.3.0)                           
##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.2.2    2016-03-24 Bioconductor                             
##  VariantAnnotation                   1.17.22  2016-03-31 Bioconductor                             
##  whisker                           * 0.3-2    2013-04-28 CRAN (R 3.3.0)                           
##  XML                                 3.98-1.4 2016-03-01 CRAN (R 3.3.0)                           
##  xtable                              1.8-2    2016-02-05 CRAN (R 3.3.0)                           
##  XVector                             0.11.7   2016-02-13 Bioconductor                             
##  yaml                                2.1.13   2014-06-12 CRAN (R 3.3.0)                           
##  zlibbioc                            1.17.1   2016-03-19 Bioconductor</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
