## Find the number of packages that depend/import/suggest GenomicRanges

depends <- strsplit("AllelicImbalance, AneuFinder, annmap, AnnotationHubData, Basic4Cseq, BasicSTARRseq, baySeq, biomvRCNS, BiSeq, BSgenome, bsseq, BubbleTree, bumphunter, CAFE, casper, cheung2010, chimera, ChIPComp, ChIPpeakAnno, ChIPQC, chipseq, chromPlot, CINdex, cleanUpdTSeq, cn.mops, CNPBayes, cnvGSA, CNVPanelizer, compEpiTools, consensusSeekeR, CSAR, csaw, DASiR, deepSNV, DESeq2, DEXSeq, DiffBind, diffHic, DMRcaller, DMRcatedata, DMRforPairs, DNAshapeR, DOQTL, EatonEtAlChIPseq, EnrichedHeatmap, ensembldb, ensemblVEP, epigenomix, exomeCopy, fastseg, FourCSeq, GeneBreak, geneRxCluster, GenoGAM, GenomicAlignments, GenomicFeatures, GenomicFiles, GenomicTuples, genoset, geuvStore, geuvStore2, gmapR, GMRP, GOTHiC, GreyListChIP, groHMM, gtrellis, GUIDEseq, Guitar, Gviz, hiAnnotator, HilbertCurve, HiTC, htSeqTools, IdeoViz, InPAS, intansv, isomiRs, MBASED, metagene, methyAnalysis, methylPipe, minfi, MMDiff, OmicCircos, PGA, PING, podkat, QuasR, r3Cseq, Rariant, Rcade, recoup, regioneR, rfPred, rGREAT, riboSeqR, RIPSeeker, RnBeads, RnBeads.hg19, RnBeads.hg38, RnBeads.mm10, RnBeads.mm9, RnBeads.rn5, Rsamtools, RSVSim, rtracklayer, segmentSeq, seqbias, SGSeq, SigFuge, SNPhood, SNPlocs.Hsapiens.dbSNP.20090506, SNPlocs.Hsapiens.dbSNP.20100427, SNPlocs.Hsapiens.dbSNP.20101109, SNPlocs.Hsapiens.dbSNP.20110815, SNPlocs.Hsapiens.dbSNP.20111119, SNPlocs.Hsapiens.dbSNP.20120608, SNPlocs.Hsapiens.dbSNP141.GRCh38, SNPlocs.Hsapiens.dbSNP142.GRCh37, SomatiCA, SomaticSignatures, SummarizedExperiment, TarSeqQC, trackViewer, TransView, traseR, VanillaICE, VariantAnnotation, VariantTools, vtpnet, wavClusteR, XtraSNPlocs.Hsapiens.dbSNP141.GRCh38, yriMulti", ", ")[[1]]

imports <- strsplit("ALDEx2, ArrayExpressHTS, BadRegionFinder, ballgown, bamsignals, beadarray, BEAT, biovizBase, BiSeq, BSgenome, CAGEr, CexoR, cgdv17, ChAMP, chipenrich, chipenrich.data, ChIPseeker, chipseq, ChIPseqR, chromDraw, CNEr, CNPBayes, coMET, contiBAIT, conumee, copynumber, CopywriteR, COSMIC.67, CoverageView, CrispRVariants, customProDB, DChIPRep, debrowser, DEFormats, derfinder, derfinderPlot, DMRcate, DRIMSeq, easyRNASeq, EDASeq, ELMER, ELMER.data, epivizr, erma, flipflop, FourCSeq, FunciSNP, genomation, genomeIntervals, GenomicAlignments, GenomicInteractions, genotypeeval, GenVisR, GGBase, ggbio, GGtools, GoogleGenomics, gQTLBase, gQTLstats, gwascat, h5vc, hiReadsProcessor, HTSeqGenie, INSPEcT, IVAS, JunctionSeq, LedPred, leeBamViews, LOLA, lumi, M3D, MafDb.ALL.wgs.phase1.release.v3.20101123, MafDb.ALL.wgs.phase3.release.v5a.20130502, MafDb.ALL.wgs.phase3.release.v5b.20130502, MafDb.ESP6500SI.V2.SSA137, MafDb.ExAC.r0.3.sites, MEAL, MEDIPS, methyAnalysis, MethylSeekR, methylumi, MinimumDistance, MMDiff, mosaics, motifbreakR, MultiAssayExperiment, NarrowPeaks, nucleR, oligoClasses, OrganismDbi, Pbase, pepDat, pepStat, phastCons100way.UCSC.hg19, phastCons100way.UCSC.hg38, phastCons7way.UCSC.hg38, PICS, prebs, proBAMr, Pviz, pwOmics, QDNAseq, qpgraph, R3CPET, R453Plus1Toolbox, RareVariantVis, rCGH, regioneR, regionReport, Repitools, RiboProfiling, RNAprobR, rnaSeqMap, roar, seq2pathway, SeqArray, seqPattern, seqplots, SeqVarTools, ShortRead, SICtools, simulatorZ, SMITE, SNPchip, SNPlocs.Hsapiens.dbSNP.20090506, SNPlocs.Hsapiens.dbSNP.20100427, SNPlocs.Hsapiens.dbSNP.20101109, SNPlocs.Hsapiens.dbSNP.20110815, SNPlocs.Hsapiens.dbSNP.20111119, SNPlocs.Hsapiens.dbSNP.20120608, SNPlocs.Hsapiens.dbSNP141.GRCh38, SNPlocs.Hsapiens.dbSNP142.GRCh37, SNPlocs.Hsapiens.dbSNP144.GRCh37, SNPlocs.Hsapiens.dbSNP144.GRCh38, soGGi, SomatiCA, SomaticCancerAlterations, spliceR, SplicingGraphs, STAN, SVM2CRM, systemPipeR, TCGAbiolinks, TFBSTools, tracktables, transcriptR, triplex, VariantFiltering, waveTiling, XtraSNPlocs.Hsapiens.dbSNP141.GRCh38, XtraSNPlocs.Hsapiens.dbSNP144.GRCh37, XtraSNPlocs.Hsapiens.dbSNP144.GRCh38", ", ")[[1]]

suggests <- strsplit("AnnotationHub, BeadArrayUseCases, biobroom, BiocGenerics, BiocParallel, Chicago, cummeRbund, GenomeInfoDb, GeuvadisTranscriptExpr, HDF5Array, interactiveDisplay, IRanges, metaseqR, MiRaGE, NarrowPeaks, NGScopy, RTCGA, S4Vectors, SeqGSEA, Single.mTEC.Transcriptomes", ", ")[[1]]

df <- data.frame(package = c(depends, imports, suggests), type = rep(c('depends', 'imports', 'suggests'), c(length(depends), length(imports), length(suggests))))

## Remove repeated
length(unique(df$package))
## April 1st 2016 from Bioc-devel: 286
