## Find the number of packages that depend/import/suggest GenomicRanges

depends <- strsplit("airway, AllelicImbalance, annmap, Basic4Cseq, baySeq, biomvRCNS, BiSeq, BSgenome, bsseq, bumphunter, CAFE, casper, cheung2010, chimera, ChIPQC, chipseq, cleanUpdTSeq, cn.mops, compEpiTools, COPDSexualDimorphism, CSAR, csaw, DASiR, deepSNV, DESeq2, DEXSeq, DiffBind, DMRforPairs, DREAM4, dsQTL, EatonEtAlChIPseq, ensemblVEP, epigenomix, epivizr, exomeCopy, fastseg, fission, FourCSeq, geneRxCluster, genomes, GenomicAlignments, GenomicFeatures, GenomicFiles, GenomicTuples, genoset, GenoView, ggtut, gmapR, GOTHiC, groHMM, Gviz, hiAnnotator, HiTC, htSeqTools, IdeoViz, intansv, MBASED, metagene, methyAnalysis, methylPipe, minfi, MMDiff, OmicCircos, parathyroidSE, PING, QuasR, r3Cseq, R453Plus1Toolbox, Rariant, Rcade, rfPred, riboSeqR, RIPSeeker, Rsamtools, RSVSim, rtracklayer, segmentSeq, seqbias, SGSeq, SigFuge, SNPlocs.Hsapiens.dbSNP.20090506, SNPlocs.Hsapiens.dbSNP.20100427, SNPlocs.Hsapiens.dbSNP.20101109, SNPlocs.Hsapiens.dbSNP.20110815, SNPlocs.Hsapiens.dbSNP.20111119, SNPlocs.Hsapiens.dbSNP.20120608, SNPlocs.Hsapiens.dbSNP141.GRCh38, SomatiCA, SomaticSignatures, SplicingGraphs, trackViewer, TransView, VanillaICE, VariantAnnotation, VariantTools, vtpnet, wavClusteR", ", ")[[1]]

imports <- strsplit("ALDEx2, AnnotationHub, ArrayExpressHTS, ballgown, beadarray, BEAT, biovizBase, BiSeq, BSgenome, CAGEr, CexoR, chipenrich, chipenrich.data, ChIPseeker, chipseq, ChIPseqR, CNEr, copynumber, COSMIC.67, CoverageView, customProDB, derfinder, derfinderPlot, DNaseR, DOQTL, easyRNASeq, flipflop, FourCSeq, FunciSNP, GenomicAlignments, GenomicInteractions, GGBase, ggbio, GGtools, gwascat, h5vc, HTSeqGenie, HTSFilter, leeBamViews, lumi, M3D, MafDb.ALL.wgs.phase1.release.v3.20101123, MafDb.ALL.wgs.phase1.release.v3.20101123, MafDb.ESP6500SI.V2.SSA137.dbSNP138, MafDb.ESP6500SI.V2.SSA137.dbSNP138, MEDIPS, methyAnalysis, MethylSeekR, methylumi, MinimumDistance, MMDiff, NarrowPeaks, nucleR, oligoClasses, pepDat, pepStat, phastCons100way.UCSC.hg19, phastCons100way.UCSC.hg19, PICS, prebs, proBAMr, Pviz, qpgraph, QuasR, regionReport, Repitools, rnaSeqMap, roar, SeqArray, seqplots, SeqVarTools, ShortRead, simulatorZ, SNPchip, SNPlocs.Hsapiens.dbSNP.20090506, SNPlocs.Hsapiens.dbSNP.20100427, SNPlocs.Hsapiens.dbSNP.20101109, SNPlocs.Hsapiens.dbSNP.20110815, SNPlocs.Hsapiens.dbSNP.20111119, SNPlocs.Hsapiens.dbSNP.20120608, SNPlocs.Hsapiens.dbSNP141.GRCh38, SomatiCA, SomaticCancerAlterations, spliceR, SplicingGraphs, TFBSTools, ToPASeq, tracktables, triplex, VariantFiltering, waveTiling", ", ")[[1]]

suggests <- strsplit("BeadArrayUseCases, BiocGenerics, BiocParallel, cummeRbund, DMRcate, GenomeInfoDb, interactiveDisplay, IRanges, metaseqR, MiRaGE, NarrowPeaks, NGScopy, SeqGSEA, STAN", ", ")[[1]]

df <- data.frame(package = c(depends, imports, suggests), type = rep(c('depends', 'imports', 'suggests'), c(length(depends), length(imports), length(suggests))))

## Remove repeated
length(unique(df$package))
